pkg/agent/agent.go

```diff
@@ -239,10 +240,25 @@ func InitializeProfiling(cpuProfile, memProfile string) {
 	}
 }
 
-func FinalizeInitialization() error {
+func SetFinalizeAgentFunc(f func() error) {
+	agent.finalizeAgentInit = f
+}
+
+func finalizeInitialization() error {
+	if agent.isInitialized {
+		return nil
+	}
+
 	err := registerExternalIDPs()
 	if err != nil {
-		logger.WithError(err).Error("failed to register CRDs for external IdP config")
+		logger.WithError(err).Fatal("failed to register CRDs for external IdP config")
```

Comment:
This registration only happen on init, if we can not register all the IDPs then the agent should stop. This will avoid cases where a failed IDP registration causes a CRD to be modified.